soda/core/tests/data_source/test_profile_columns.py

```diff
@@ -114,24 +119,46 @@ def test_profile_columns_text(data_source_fixture: DataSourceFixture):
     }
 
 
-@pytest.mark.skip(reason="TODO: table names are not handled for some of the data sources")
-# @pytest.mark.skipif(
-#     test_data_source == "spark_df",
-#     reason="TODO: fix for spark_df.",
-# )
 def test_profile_columns_all_tables_all_columns(data_source_fixture: DataSourceFixture):
-    _ = data_source_fixture.ensure_test_table(customers_profiling)
     scan = data_source_fixture.create_test_scan()
-    mock_soda_cloud = scan.enable_mock_soda_cloud()
+
     scan.add_sodacl_yaml_str(
         """
             profile columns:
-                columns: ["%.%"]
+                columns:
+                  - "%.%"
         """
     )
-    scan.execute(allow_warnings_only=True)
-    profiling_result = mock_soda_cloud.pop_scan_result()
-    assert len(profiling_result["profiling"]) >= 1
+    data_source_name = data_source_fixture.data_source_name
+    data_source_scan = scan._get_or_create_data_source_scan(data_source_name)
+    data_source = data_source_scan.data_source
+    profiling_cfg = scan._sodacl_cfg.data_source_scan_cfgs[data_source_name].data_source_cfgs[0]
+    include_columns = profiling_cfg.include_columns
+    exclude_columns = profiling_cfg.exclude_columns
+
+    profile_columns_run = ProfileColumnsRun(data_source_scan, profiling_cfg)
+
+    table_names: dict[str, int] = data_source.get_table_names(
+        include_tables=profile_columns_run._get_table_expression(include_columns),
+        exclude_tables=profile_columns_run._get_table_expression(exclude_columns, is_for_exclusion=True),
+        query_name="profile-columns-get-table-names",
+    )
+    assert len(table_names) > 2
+
+    parsed_included_tables_and_columns = profile_columns_run._build_column_expression_list(include_columns)
+    assert parsed_included_tables_and_columns == {"%": ["%"]}
+
+    parsed_excluded_tables_and_columns = profile_columns_run._build_column_expression_list(exclude_columns)
+    assert parsed_excluded_tables_and_columns == {}
+
+    for table_name in table_names:
+        columns_metadata_result = data_source.get_table_columns(
+            table_name=table_name,
+            query_name=f"profile-columns-get-column-metadata-for-{table_name}",
+            included_columns=parsed_included_tables_and_columns,
+            excluded_columns=parsed_excluded_tables_and_columns,
+        )
+        assert columns_metadata_result
```

Comment:
Does this just assert that there is at least one column (i.e. that the object is not None, but would we want to/can we have a more expected content to assert maybe?